src/Cake.Prca/Orchestrator.cs

```diff
@@ -0,0 +1,270 @@
+﻿namespace Cake.Prca
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+    using Core.Diagnostics;
+    using Issues;
+    using PullRequests;
+
+    /// <summary>
+    /// Class for writing code analysis issues to pull requests.
+    /// </summary>
+    internal class Orchestrator
+    {
+        private readonly ICakeLog log;
+        private readonly ICodeAnalysisProvider codeAnalysisProvider;
+        private readonly IPullRequestSystem pullRequestSystem;
+        private readonly ReportCodeAnalysisIssuesToPullRequestSettings settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Orchestrator"/> class.
+        /// </summary>
+        /// <param name="log">Cake log instance.</param>
+        /// <param name="codeAnalysisProvider">Code analysis issue provider.</param>
+        /// <param name="pullRequestSystem">Object for accessing pull request system.</param>
+        /// <param name="settings">Settings.</param>
+        public Orchestrator(
+            ICakeLog log,
+            ICodeAnalysisProvider codeAnalysisProvider,
+            IPullRequestSystem pullRequestSystem,
+            ReportCodeAnalysisIssuesToPullRequestSettings settings)
+        {
+            log.NotNull(nameof(log));
+            codeAnalysisProvider.NotNull(nameof(codeAnalysisProvider));
+            pullRequestSystem.NotNull(nameof(pullRequestSystem));
+            settings.NotNull(nameof(settings));
+
+            this.log = log;
+            this.codeAnalysisProvider = codeAnalysisProvider;
+            this.pullRequestSystem = pullRequestSystem;
+            this.settings = settings;
+        }
+
+        /// <summary>
+        /// Runs the orchestrator.
+        /// Posts new issues, ignoring duplicate comments and resolves comments that were open in an old iteration
+        /// of the pull request.
+        /// </summary>
+        public void Run()
+        {
+            var issues = this.codeAnalysisProvider.ReadIssues().ToList();
+
+            this.ValidateIssues(issues);
+            this.log.Information("Processing {0} new issues", issues.Count);
+
+            this.PostAndResolveComments(issues);
+        }
+
+        /// <summary>
+        /// Validate issues if they can be processed.
+        /// </summary>
+        /// <param name="issues">Issues to validate.</param>
+        private void ValidateIssues(
+            IEnumerable<ICodeAnalysisIssue> issues)
+        {
+            issues.NotNull(nameof(issues));
+
+            // TODO
```

Comment:
@stffabi The question here is, is there something that we should validate here? I think we can get rid of the method now, since we already check stuff in the `CodeAnalysisIssue` ctor.